pkg/resource/notebook_instance/testdata/test_suite.yaml

```diff
@@ -0,0 +1,319 @@
+          latest_state: "notebook_instance/v1alpha1/input/update/nb_update_latest_inservice_input.yaml"
+          svc_api:
+            - operation: StopNotebookInstance
+        invoke: Update
+        expect:
+          latest_state: "notebook_instance/v1alpha1/output/nb_update_inservice.yaml"
+          error: "NotebookInstance is stopping."
+      - name: "Update=Stopped--ByController"
+        description: "This test checks if the controller updates and sets the StoppedByControllerMetadata status."
+        given:
+          desired_state: "notebook_instance/v1alpha1/input/update/nb_update_desired_common.yaml"
+          latest_state: "notebook_instance/v1alpha1/input/update/nb_update_latest_stopped_controller_input.yaml"
+          svc_api:
+            - operation: UpdateNotebookInstanceWithContext 
+        invoke: Update
+        expect:
+          latest_state: "notebook_instance/v1alpha1/output/nb_update_stopped_controller.yaml"
+      - name: "Update=Stopped--ByControllerUser"
+        description: "This test checks if the controller updates and does not set the StoppedByControllerMetadata status."
+        given:
+          desired_state: "notebook_instance/v1alpha1/input/update/nb_update_desired_common.yaml"
+          latest_state: "notebook_instance/v1alpha1/input/update/nb_update_latest_stopped_by_user_input.yaml"
+          svc_api:
+            - operation: UpdateNotebookInstanceWithContext 
+        invoke: Update
+        expect:
+          latest_state: "notebook_instance/v1alpha1/output/nb_update_stopped_by_user.yaml"
+      - name: "Update=InvalidInput"
+        description: "This test checks if invalid input has been entered."
+        given:
+          desired_state: "notebook_instance/v1alpha1/input/update/nb_update_desired_invalid.yaml"
+          latest_state: "notebook_instance/v1alpha1/input/update/nb_update_latest_stopped_by_user_input.yaml"
+          svc_api:
+            - operation: UpdateNotebookInstanceWithContext
+              error:
+                code: ValidationException
+                message: "VolumeInSizeGB cannot be lower than current."
+        invoke: Update
+        expect:
+          latest_state: "notebook_instance/v1alpha1/output/nb_update_invalid.yaml"
+      - name: "Update=Dissassociated"
+        description: "This test checks if the dissassociated parameters are being handled right."
+        given:
+          desired_state: "notebook_instance/v1alpha1/input/update/nb_update_desired_common.yaml"
+          latest_state: "notebook_instance/v1alpha1/input/update/nb_update_latest_stopped_dissassociated.yaml"
+          svc_api:
+            - operation: UpdateNotebookInstanceWithContext
+        invoke: Update
+        expect:
+          latest_state: "notebook_instance/v1alpha1/output/nb_update_dissassociated.yaml"
+
+  #TODO: Add tests for Stop errors once unit test supports that. Also replace status with annotation in the following files:
```

Comment:
what changes are needed for this? framework already supports throwing error from APIs

cant you do something similar to the `Create=InvalidInput` test?

Reply:
I'm not exactly sure, for some reason I cant get an error to return for Start/StopNotebookInstance. I inserted a print statement here(https://github.com/aws-controllers-k8s/sagemaker-controller/blob/cd9c3be74768a64f25fb67e796f83a481643ca16/pkg/resource/notebook_instance/custom_sdk_api.go#L28) that printed the error and it printed nil on the test where StartNotebookInstance was supposed to return an error.

Reply:
Nevermind it seems to work, might have had a typo when I tried it before.